src/app/event/components/UserTicket.tsx

```diff
@@ -13,6 +13,40 @@ export type TicketInfoType =
     }
   | undefined;
 
+const ButtonCopyToClipBoard = ({
+  contentToCopy,
+  buttonText,
+  className,
+}: {
+  contentToCopy: string;
+  buttonText: string;
+  className: string;
+}) => {
+  const [copied, setCopied] = useState(false);
+
+  const copyToClipboard = async (text: string) => {
+    await navigator.clipboard.writeText(text);
+    setCopied(true);
+    setTimeout(() => {
+      setCopied(false);
+    }, 2000);
+  };
```

Comment:
good candidate for a hook. I think we probably have one in fact to do this in @bleu/ui